core/app/models/spree/stock_location.rb

```diff
@@ -99,6 +101,16 @@ def fill_status(variant, quantity)
       end
     end
 
+    def self.default
+      Rails.cache.fetch("#{Rails.application.class.parent_name.underscore}_default_stock_location") do
+        default_location = where(default: true).first
+        if default_location.blank? && (default_location = first)
+          default_location.update(default: true)
```

Comment:
Instead of fixing data on first access, what about providing a migration that fixes the data once instead of maintaining a data driven branch that on most installations will only be executed once, so is for all but the first iteration dead code.

Ideally push the "a default exists" constraint down to the DB.